api/src/pcapi/core/finance/api.py

```diff
@@ -2909,25 +2928,56 @@ def _has_celebrated_birthday_since_credit_or_registration(user: users_models.Use
     return first_registration_datetime.date() < latest_birthday_date
 
 
-def _has_been_recredited(user: users_models.User) -> bool:
-    if user.age is None:
+def _has_been_recredited(user: users_models.User, age: int | None = None) -> bool:
+    if age is None:
+        age = user.age
+
+    if age is None:
         logger.error("Trying to check recredit for user that has no age", extra={"user_id": user.id})
         return False
 
     if user.deposit is None:
         return False
 
     known_age_at_deposit = _get_known_age_at_deposit(user)
-    if known_age_at_deposit == user.age:
+    if known_age_at_deposit == age:
         return True
 
     if len(user.deposit.recredits) == 0:
         return False
 
-    return conf.RECREDIT_TYPE_AGE_MAPPING[user.age] in [recredit.recreditType for recredit in user.deposit.recredits]
+    has_been_recredited = conf.RECREDIT_TYPE_AGE_MAPPING[age] in [
+        recredit.recreditType for recredit in user.deposit.recredits
+    ]
+    if has_been_recredited:
+        return True
+
+    if age == 16:
```

Comment:
On n'a pas une constante ou une fonction de validation pour ça ? J'ai l'impression qu'on d'habitude des constantes pour les types de crédits, ça permettrait peut-être d'avoir un ensemble homogène et d'y voir plus clair.

Mais c'est vraiment du détail.

Reply:
Malheureusement, les dictionnaires définis dans `finance/conf.py` utilisent directement les âges `15`, `16` et `17` comme clés...

Comme ce cas limite est très spécifique à l'âge `16`, je ne suis pas sûr qu'on puisse avoir un gain d'homogénéité ici